caddyhttp/httpserver/server.go

```diff
@@ -507,16 +507,35 @@ func (s *Server) Stop() error {
 // OnStartupComplete lists the sites served by this server
 // and any relevant information, assuming caddy.Quiet == false.
 func (s *Server) OnStartupComplete() {
-	if caddy.Quiet {
-		return
+	if !caddy.Quiet {
+		firstSite := s.sites[0]
+		scheme := "HTTP"
+		if firstSite.TLS.Enabled {
+			scheme = "HTTPS"
+		}
+		fmt.Println("")
+		fmt.Printf("Serving %s on port "+firstSite.Port()+" \n", scheme)
+		s.OutputSiteInfo("fmt")
+	}
+	// if caddy process log is going to stdout, printing to log
+	// here would duplicate to stdout so dont.  If -quiet ensure
+	// that Log is still output even if process log startup isnt. #2469
+	if caddy.LogDestination != "stdout" || caddy.Quiet {
+		s.OutputSiteInfo("log")
 	}
+}
+
+func (s *Server) OutputSiteInfo(LogType string) {
 	for _, site := range s.sites {
 		output := site.Addr.String()
 		if caddy.IsLoopback(s.Address()) && !caddy.IsLoopback(site.Addr.Host) {
 			output += " (only accessible on this machine)"
 		}
-		fmt.Println(output)
-		log.Println(output)
+		if LogType == "fmt" {
```

Comment:
What is "fmt"?

Reply:
It was just a way of differntiating between log outputs.  I'm not sure what name to give it to maybe "log" and "console" or something?

Reply:
What do you mean by "log outputs" though? It looks like you're using it more look a boolean, like whether the log is configured to write to stdout specifically. Did I get that right?

Reply:
At the moment, it's a binary option. A boolean would be an adequate representation. Leaving it as a string allows for more options for handling to be added in future, although I can't imagine what other options there'd be for this function.